ibazel/source_event_handler.go

```diff
@@ -1,47 +0,0 @@
-// Copyright 2017 The Bazel Authors. All rights reserved.
-//
-// Licensed under the Apache License, Version 2.0 (the "License");
-// you may not use this file except in compliance with the License.
-// You may obtain a copy of the License at
-//
-//    http://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
-
-package main
-
-import (
-	"github.com/bazelbuild/bazel-watcher/ibazel/fs_watcher"
-)
-
-type SourceEventHandler struct {
-	SourceFileEvents  chan fs_watcher.Event
-	SourceFileWatcher fs_watcher.Watcher
-}
-
-func (s *SourceEventHandler) Listen() {
-	for {
-		select {
-		case event := <-s.SourceFileWatcher.Events():
-			s.SourceFileEvents <- event
-
-			switch event.Op {
-			case fs_watcher.Remove, fs_watcher.Rename:
-				s.SourceFileWatcher.Add(event.Name)
```

Comment:
@achew22 This file was deleted. As far as I could tell, this was created back when Ibazel watched individual files directly instead of their parent dirs. Now that it watches the parent dirs, I think the concern of IDEs deleting and replacing files is no longer valid. In my testing, it appeared to work correctly.